c/check.py

```diff
@@ -608,22 +623,14 @@ def main():
         for entry in entries if entry.endswith(".set")
         for pattern in read_set_file(os.path.join(main_directory, entry)))
     all_patterns = re.compile("^(" + "|".join(all_patterns_re) + ")$")
-    all_matched_files = set()
-
-    ok = True
-    for entry in entries:
-        path = os.path.join(main_directory, entry)
-        if not (entry[0] == '.' or entry == "bin" or entry.endswith("-todo")):
-            if os.path.isdir(path) and not entry in IGNORED_DIRECTORIES:
-                ok &= DirectoryChecks(path, all_patterns, entry).run()
-            elif entry.endswith(".set"):
-                check = SetFileChecks(path, entry)
-                all_matched_files.update(check.matched_files)
-                ok &= check.run()
-            else:
-                logging.debug("%s: skipped", entry)
-        else:
-            logging.debug("%s: skipped", entry)
+
+    check_func = partial(
+        _check_benchmark_entry, main_directory=main_directory, all_patterns=all_patterns
+    )
+    with Pool(4) as p:
```

Comment:
Should be make this depend on the machine? Or configurable? The latter would also provide a way to create deterministic output if a developer prefers this locally.

Reply:
I've added an optional command-line option `--num-processes`.

Example usage:

`c/check.py --num-processes=1`

Is this fine with you?